spec/takeaway_spec.rb

```diff
@@ -0,0 +1,107 @@
+require 'takeaway'
+
+describe Takeaway do
+  let(:menu) { double :menu }
+  let(:order) { double :order }
+  let(:sms_sender) { double :smssender }
+  subject(:takeaway) { described_class.new(menu, order, sms_sender) }
+ 
+  it 'contains a @menu that is an instance of Menu' do
+    ta = described_class.new()
+    expect(ta.menu).to be_an_instance_of(Menu)
+  end
+
+  describe '#show_menu' do
+    it 'shows menu using menu.show_menu' do
+      default_return = "SHOULD PRINT THIS"
+      allow(menu).to receive(:show_menu).and_return(default_return)
+      expect(takeaway.read_menu).to eq(default_return)
+    end
+  end
+
+  describe '#order_item' do
```

Comment:
There's some nice doubling here, which obviously required some thought.  I think, however, that you only need to ensure that the instance of Takeaway sends the correct messages (since managing the basket is the responsibility of Order).  The rest will be testing in other places.